pkg/check/chunkrepair/chunkrepair.go

```diff
@@ -0,0 +1,281 @@
+package chunkrepair
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/random"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// Options represents chunk repair check options
+type Options struct {
+	NumberOfChunksToRepair int
+	Seed                   int64
+}
+
+// Check ...
+func Check(c bee.Cluster, o Options, pusher *push.Pusher, pushMetrics bool) (err error) {
+	ctx := context.Background()
+	rnds := random.PseudoGenerators(o.Seed, o.NumberOfChunksToRepair)
+	fmt.Printf("Seed: %d\n", o.Seed)
+
+	pusher.Collector(repairedCounter)
+	pusher.Collector(repairedTimeGauge)
+	pusher.Collector(repairedTimeHistogram)
+
+	pusher.Format(expfmt.FmtText)
+
+	for i := 0; i < o.NumberOfChunksToRepair; i++ {
+		// Pick node A, B, C and a chunk which is closest to B
+		nodeA, nodeB, nodeC, chunk, err := getNodes(ctx, c, rnds[i])
+		if err != nil {
+			return err
+		}
+		addressA, err := nodeA.Overlay(ctx)
+		if err != nil {
+			return err
+		}
+
+		// upload the chunk in nodeA
+		err = nodeA.UploadChunk(ctx, chunk)
+		if err != nil {
+			return err
+		}
+
+		for {
+			// check if the node is there in the local store of node B
+			// this does a get chunk instead of Has chunk, so the following
+			// call just checks if the chunk is accessible from nodeB
+			present, err := nodeB.HasChunk(ctx, chunk.Address())
+			if err != nil {
+				fmt.Println("sleeping...")
```

Comment:
This could print a lot of `sleeping...` in terminal, very fast. Could this retry be improved and limited?

Reply:
not much for a cluster of our size... but i think this print is superfluous and can be removed.

Reply:
Well, this can be an endless loop regardless of the print.